javascript/ql/src/semmle/javascript/Expr.qll

```diff
@@ -488,30 +647,104 @@ class Property extends @property, Documentable {
   Decorator getADecorator() { result = getDecorator(_) }
 }
 
-/** A value property in an object literal. */
+/**
+ * A value property definition in an object literal.
+
+ * Examples:
+ *
+ * ```
+ * var p = {
+ *   x: 1,                                                          // value property
+ *   y: 1,                                                          // value property
+ *   diag: function() { return this.x - this.y; },                  // value property
+ * }
+ * ```
+ */
 class ValueProperty extends Property, @value_property { }
 
-/** A property getter or setter in an object literal. */
+/**
+ * A property getter or setter in an object literal.
+ *
+ * Examples:
+ *
+ * ```
+ * var p = {
+ *   x: 1,
+ *   y: 1,
+ *   diag: function() { return this.x - this.y; },
+ *   get area() { return this.x * this.y; },                        // property getter
+ *   set area(a) { this.x = Math.sqrt(a); this.y = Math.sqrt(a); }  // property setter
+ * }
+ * ```
+ */
 class PropertyAccessor extends Property, @property_accessor {
   override FunctionExpr getInit() { result = Property.super.getInit() }
 }
 
-/** A property getter in an object literal. */
+/**
+ * A property getter in an object literal.
+ *
+ * Example:
+ *
+ * ```
+ * var p = {
+ *   x: 1,
+ *   y: 1,
+ *   diag: function() { return this.x - this.y; },
+ *   get area() { return this.x * this.y; },                        // property getter
+ *   set area(a) { this.x = Math.sqrt(a); this.y = Math.sqrt(a); }
+ * }
+ * ```
+ */
 class PropertyGetter extends PropertyAccessor, @property_getter { }
 
-/** A property setter in an object literal. */
+/**
+ * A property setter in an object literal.
+ *
+ * Example:
+ *
+ * ```
+ * var p = {
+ *   x: 1,
+ *   y: 1,
+ *   diag: function() { return this.x - this.y; },
+ *   get area() { return this.x * this.y; },
+ *   set area(a) { this.x = Math.sqrt(a); this.y = Math.sqrt(a); }  // property setter
+ * }
+ * ```
+ */
 class PropertySetter extends PropertyAccessor, @property_setter { }
 
 /**
- * A spread property in an object literal, such as `...others` in
- * `{ x: 42, ...others }`. The value of a spread property is always
+ * A spread property in an object literal.
+ *
+ * The initializer of a spread property is always
  * a `SpreadElement`.
+ *
+ * Example:
+ *
+ * ```
+ * var options = {
+ *   ...defaultOptions,  // spread property
+ *   password: pwd
+ * }
+ * ```
  */
 class SpreadProperty extends Property {
   SpreadProperty() { not exists(getNameExpr()) }
 }
 
-/** A function expression. */
+/**
+ * A (non-arrow) function expression.
+ *
+ * Examples:
+ *
+ * ```
+ * var greet =
+ *   function() {            // function expression
+ *     console.log("Hi!");
+ *   };
+ */
```

Comment:
Missing the closing triple-backticks

Also add an example with methods, since they're extracted as anonymous `FunctionExpr`.